tests/test_service_injection.py

```diff
@@ -0,0 +1,196 @@
+import copy
```

Comment:
Can we add a test to ensure that a service/dependency can have another service/dependency injected into its constructor?